schedule/scheduler.go

```diff
@@ -30,6 +30,9 @@ type TaskConfig struct {
 	// Locker is an optional lock that must be taken before the task can execute.
 	Locker Locker
 
+	// ExitOnFailure will will cause cause a fatal failure when a task stops with an error.
+	ExitOnFailure bool
```

Comment:
I'm not sure we need this flag. Setting `RestartOnFailuer` to false more or less does the same thing.

Reply:
This fix is needed for the indexer when it fails on startup (as the RestartOnFailure flag is not considered until after locks are added which is exactly where the index fails when locks could not be acquired. (https://github.com/filecoin-project/sentinel-visor/blob/master/schedule/scheduler.go#L90-L106)

Agree it's a cludgy fix and would prefer to consider `RestartOnCompletion` inside the locking logic, but this is what was delivered and is working in production right now. Propose we refactor via a separate issue?

Reply:
As discussed in chat we will remove the `ExitOnFailure` and address the problem it is trying to fix in a follow on.

Reply:
As discussed, agreed to remove this cludgy fix entirely and find a better fix for considering `RestartOnCompletion` within the locking code.